compiler/src/iree/compiler/Dialect/HAL/IR/HALOps.td

```diff
@@ -383,11 +392,12 @@ def HAL_DispatchExternOp : HAL_PureOp<"dispatch.extern", [
   );
 
   let regions = (region
-    AnyRegion:$workgroup_count
+    AnyRegion:$workgroup_count,
+    VariadicRegion<AnyRegion>:$target_regions
   );
 
   let assemblyFormat = [{
-    $export
+    $export `` `:` `` $export_ordinal
```

Comment:
Still not quite following then. After we link don't we have to update all the export ordinals? If we specify a static ordinal here, how are we expecting to resolve that later? Does this always get the ordinal you specify? What about multiple extern dispatches with the same ordinal?

Reply:
We wouldn't re-link these external dispatches, but the external dispatches themselves can be pre-linked or we could perform SPIRV binary linking (via spirv-link instead of in MLIR). At this layer we want to be able to support the full fidelity of the HAL executables and that means supporting linked executables so we can round-trip. A litmus test is whether we can take a linked SPIRV executable flatbuffer from the compiler and use that in the input.

I've reworked things so now each object target group has its own ordinal instead of having one per op. This ordinal then gets dropped right onto the `hal.executable.export` created per target group.

Reply:
I see, pre-linking makes sense, and I'd probably need to see SPIRV binary linking in action to understand the whole picture there. One per eventual export makes sense, thanks for the explanations :)